datahub-frontend/app/client/AuthServiceClient.java

```diff
@@ -97,4 +249,31 @@ private String getAccessTokenFromJson(final String jsonStr) {
       throw new IllegalArgumentException("Failed to parse JSON received from the MetadataService!");
     }
   }
+
+  private boolean getIsNativeUserCreatedFromJson(final String jsonStr) {
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      return mapper.readTree(jsonStr).get(IS_NATIVE_USER_CREATED_FIELD).asBoolean();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Failed to parse JSON received from the MetadataService!");
+    }
+  }
+
+  private boolean getAreNativeUserCredentialsResetFromJson(final String jsonStr) {
+    ObjectMapper mapper = new ObjectMapper();
+    try {
+      return mapper.readTree(jsonStr).get(ARE_NATIVE_USER_CREDENTIALS_RESET_FIELD).asBoolean();
+    } catch (Exception e) {
+      throw new IllegalArgumentException("Failed to parse JSON received from the MetadataService!");
+    }
+  }
+
+  private boolean getDoesPasswordMatchFromJson(final String jsonStr) {
```

Comment:
The whole "does password match" naming convention is pretty weird.. 

It would be simpler if GMS just said if the request to verify a password succeeded or not. 

Minor, not a blocker